pkg/apis/operator/v1/types.go

```diff
@@ -221,6 +233,13 @@ type CalicoNetworkSpec struct {
 	// +optional
 	// +kubebuilder:validation:Enum=None;Multus
 	MultiInterfaceMode *MultiInterfaceMode `json:"multiInterfaceMode,omitempty"`
+
+	// BPFDataplaneMode configures whether the BPF data plane is enabled anda in
+	// what form.
+	// Default: Disabled
+	// +optional
+	// +kubebuilder:validation:Enum=Disabled;Enabled;EnabledKeepKubeProxy
+	BPFDataplaneMode *BPFDataplaneMode `json:"bpfDataplaneMode,omitempty"`
```

Comment:
We should discuss where this option belongs. Not 100% obvious to me that this beloings in `CalicoNetworkSpec`.

Reply:
For example, we might want to do something like this:
- Always have the operator set KUBERNETES_SERVICE_PORT and KUBERNETES_SERVICE_HOST
- Users enable / disable BPF itself via FelixConfiguration resources. 

Would that work? It wouldn't require any API changes here.